codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/generators/protocol/ServerProtocolTestGenerator.kt

```diff
@@ -464,7 +465,8 @@ class ServerProtocolTestGenerator(
     private fun checkResponse(rustWriter: RustWriter, testCase: HttpMalformedResponseDefinition) {
         checkStatusCode(rustWriter, testCase.code)
         checkHeaders(rustWriter, "&http_response.headers()", testCase.headers)
-        checkHttpResponseExtensions(rustWriter)
+        // TODO Enable.
+        // checkHttpResponseExtensions(rustWriter)
```

Comment:
It should be enabled in this PR, but it currently fails for malformed request tests because I don't know what to do here https://github.com/awslabs/smithy-rs/pull/1233/files#r818735202

Reply:
In the end I had to remove this check from _all_ test types, because we now set extensions in the operation handler trait implementation, a code path that no tests exercise. See the added comments here https://github.com/awslabs/smithy-rs/pull/1233/files#diff-82b743775b074473ce5ed5ad9493413249796ca370ace0b1fa5921cb40211e28R473-R480.